service-base/src/main/java/org/eclipse/hono/service/credentials/CredentialsMessageFilter.java

```diff
@@ -46,10 +46,7 @@ public static boolean verify(final ResourceIdentifier linkTarget, final Message
 
     protected static final boolean verifySingleAmqpValueMessage(final ResourceIdentifier linkTarget, final Message msg) {
         if (msg.getBody() != null) {
-            if (msg.getContentType() == null || !msg.getContentType().startsWith("application/json")) {
-                LOG.trace("message [{}] content type is not JSON", msg.getMessageId());
-                return false;
-            } else if (!(msg.getBody() instanceof AmqpValue)) {
+            if (!(msg.getBody() instanceof AmqpValue)) {
```

Comment:
the check for 
```
if (msg.getBody() != null) {
...
```
is obsolete since it is covered by the `instanceof` check already.